packages/banner/src/react/index.tsx

```diff
@@ -66,43 +65,21 @@ const Banner = React.forwardRef((props, ref) => {
   )
 }) as BannerComponent
 
-Banner.displayName = 'Banner'
-
-interface AnchorProps extends HTMLPropsFor<'a'> {
-  href: string
-}
-interface ButtonProps extends HTMLPropsFor<'button'> {
-  href?: undefined
-}
-type ButtonComponent = React.ForwardRefExoticComponent<unknown> & {
-  (props: AnchorProps, ref?: RefFor<'a'>): JSX.Element
-  (props: ButtonProps, ref?: RefFor<'button'>): JSX.Element
-}
 const Button = React.forwardRef<
   HTMLAnchorElement | HTMLButtonElement,
-  AnchorProps | ButtonProps
+  React.ComponentProps<typeof DSButton>
 >((props, ref) => {
   const color = React.useContext(ColorContext)
-
-  return 'href' in props ? (
-    <DSButton
-      {...(props as HTMLPropsFor<'a'>)}
-      {...styles.button({ color })}
-      appearance={DSButton.appearances.stroke}
-      href={props.href || ''}
-      ref={ref as RefFor<'a'>}
-      size={DSButton.sizes.small}
-    />
-  ) : (
+  return (
     <DSButton
-      {...(props as HTMLPropsFor<'button'>)}
+      {...(props as HTMLPropsFor<HTMLButtonElement | HTMLAnchorElement>)}
       {...styles.button({ color })}
       appearance={DSButton.appearances.stroke}
-      ref={ref as RefFor<'button'>}
+      ref={ref as any}
```

Comment:
Did you pull this down? I'm pretty sure that create a TS error but I could be wrong. I worked on this way too long resolving typescript issues. 

Honestly at the end I realized I personally rather use any for ref with React. Your call though if it does not throw an error. Just remember this is wrapping a component that if and Exotic one because it's wrapped in forwardRef already.

Reply:
Yes, I pulled this down. I did not attempt to modify the code in this manner. 

I was drawn to this line because of the cast to `any`, because it signifies to me that it's a place where TS isn't working for us and we work around it. For that reason, I am currently not in favor of using an `any` cast as the default way of working with refs. But I'm open to hearing more about the reasoning that led you to that conclusion.

Reply:
It does throw a type error. I can try and refactor button to use the polymorphic util `forwardRefWithAsAndStatics` and see if we can then use this typing. 

Type 'Ref<HTMLButtonElement | HTMLAnchorElement>' is not assignable to type 'LegacyRef<HTMLButtonElement> | undefined'.
  Type 'RefObject<HTMLButtonElement | HTMLAnchorElement>' is not assignable to type 'LegacyRef<HTMLButtonElement> | undefined'.
    Type 'RefObject<HTMLButtonElement | HTMLAnchorElement>' is not assignable to type 'RefObject<HTMLButtonElement>'.
      Type 'HTMLButtonElement | HTMLAnchorElement' is not assignable to type 'HTMLButtonElement'.
        Type 'HTMLAnchorElement' is missing the following properties from type 'HTMLButtonElement': disabled, form, formAction, formEnctype, and 11 more.ts(2322)
index.d.ts(147, 9): The expected type comes from property 'ref' which is declared here on type 'IntrinsicAttributes & ButtonProps'
(JSX attribute) ref?: React.LegacyRef<HTMLButtonElement> | undefined

Reply:
Going to make this a separate issue